### DISCUSSION TITLE
Retrieve number of read/write operations for a particular file in Cephfs

### DISCUSSION THREAD

**Discussion by ltson4121994** [ceph/go-ceph#810] [Q&A] [answered]
I have been studying the document from Ceph and Rados but I could not find any metrics to measure the number of read/write operation for each file. I understand that Cephfs is the front-end, the file is going to be stored as a object in the OSD and I have found that Ceph provides Cache Tiering feature which also requires the monitor for read/write operation for each object. Could someone please give me guidance on how this is achieved? Thanks.

**Comment by phlogistonjohn** [comment, marked as answer]
Interesting question. It may be helpful to ask this question on one of the ceph mailing lists as I'm unaware of any existing API to get these stats. You may also want to clarify if this is a global count or a "per session" (per rados/cephfs "connection") counter.  If these values are available somewhere, but unknown to me, we can discuss adding them to go-ceph, which typically just wraps existing APIs provided by Ceph (either C apis or mgr based APIs). If the counters don't exist you can then ask the ceph project for a feature enhancement and if the feature gets added we can then add it to our Go bindings.

I would ask in ceph users or devel mailing lists: https://ceph.io/en/community/connect/

**Reply by ltson4121994** [reply]
Thank you very much for your reply. I will ask this question in the ceph mailing list.
